llama_cpp/llama_chat_format.py

```diff
@@ -783,9 +785,30 @@ def functionary_chat_handler(
     model: Optional[str] = None,
     logits_processor: Optional[llama.LogitsProcessorList] = None,
     grammar: Optional[llama.LlamaGrammar] = None,
+    hf_tokenizer_path: Optional[str] = None,
     **kwargs,  # type: ignore
 ) -> Union[llama_types.ChatCompletion, Iterator[llama_types.ChatCompletionChunk]]:
     SYSTEM_MESSAGE = """A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. The assistant calls functions with appropriate input when necessary"""
+    
```

Comment:
I know the original functionary model is deprecated but I want to breaking any code that's using the old model, can we introduce these chat formats as `functionary-v1` and `funtionary-v2` for the new models.

Reply:
I think the code already checked the model version to use the proper chat template

Reply:
I see that you have a branch that keeps the old v1 code. I will just copy those relevant parts from there into my branch and deduplicate as much code as possible between the new and old chat handlers